### PR TITLE
Add Basic with CSRF scenario to the App Gateway tests

### PR DESCRIPTION
**Pull request by franpog859**
**Description**

Changes proposed in this pull request:

- Add Basic with CSRF scenario to the App Gateway tests

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/14654

### CODE REVIEW

File: tests/components/application-connector/resources/charts/gateway-test/templates/basic-cases.yml [hunk 1]
```diff
@@ -30,6 +30,48 @@ spec:
...
+      entries:
+        - type: API
+          targetUrl: "http://test-app.{{ .Values.namespace  }}.svc.cluster.local:8080/v1/api/csrf-basic/ok"
+          centralGatewayUrl: "http://central-application-gateway.kyma-system:8080/test-app/csrf-basic"
+          credentials:
+            secretName: basic-test
+            type: Basic
+            csrfInfo:
+              tokenEndpointURL: "http://test-app.{{ .Values.namespace  }}.svc.cluster.local:8080/v1/api/csrf/token"
+    - displayName: bad csrf token basic
```
Comment (mvshao) [inline tests/components/application-connector/resources/charts/gateway-test/templates/basic-cases.yml:47]: In my opinion, separating positive and negative cases into separate Application CRD will increase the readability of the tests.

Comment (franpog859) [reply tests/components/application-connector/resources/charts/gateway-test/templates/basic-cases.yml:47]: Hmm, I'm not sure as we probably will have only one or two positive cases per authorization method (so also per an Application) and the cases are going to be quite well described with the return codes. If you still think we should separate it, I'll do it after the rebase onto main with the PR that adds the multiple Application map to the test suite, @mvshao

File: tests/components/application-connector/internal/testkit/test-api/csrf.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+				handleError(w, http.StatusForbidden, "CSRF token header missing")
+				return
+			}
+
+			ch.mutex.RLock()
+			_, found := ch.tokens[headerToken]
+			ch.mutex.RUnlock()
+
+			if !found {
+				handleError(w, http.StatusForbidden, "Invalid token from the header")
```
Comment (VOID404) [inline tests/components/application-connector/internal/testkit/test-api/csrf.go:71]:
```suggestion
				handleError(w, http.StatusForbidden, "Invalid CSRF token from the header")
```

**Review by VOID404 — CHANGES_REQUESTED** [review]

**Review by VOID404 — APPROVED** [review]

**Pull request merged**